server/src/main/java/org/apache/iotdb/db/engine/storagegroup/DataRegion.java

```diff
@@ -2069,6 +2249,235 @@ private List<TsFileResource> getFileResourceListForQuery(
     return tsfileResourcesForQuery;
   }
 
+  /**
+   * alter timeseries encoding & compressionType<br>
+   * 1、flush and close tsfile<br>
+   * 2、locks<br>
+   * 3、write temp tsfiles<br>
+   * 4、unregister old tsfiles and release locks<br>
+   * 5、rename temp tsfiles<br>
+   * 6、register tsfiles<br>
+   */
+  public void alter(
+      PartialPath fullPath, TSEncoding curEncoding, CompressionType curCompressionType)
```

Comment:
What if fullPath is in the following format: root.sg.d1.**? Missing path judgment with wildcard characters.

Reply:
fullPath
    : ROOT (DOT nodeNameWithoutWildcard)*
    ;
FullPath does not support wildcards. There is no general method for determining wildcards. Do you need to add a wildcard method?